In [2]:
# Import necessary libraries

import pandas as pd

In [7]:
# Load the csv file

prompts_clients_df = pd.read_csv("data/prompts_clients.csv")
prompts_clients_df.head()

,client_id,county,sub_county,facility_name,gestation_weeks,parity,age,registration_date,interaction_date,interaction_type,...,referral_recommended,referral_destination,time_to_facility_hours,visited_facility,visit_outcome,delivery_outcome,mode_of_delivery,maternal_complication,client_experience_rating,experience_comment_theme
0,C00001,Kiambu,Ruiru,Ruiru Sub-County Hospital,14,0,34,2024-09-07,2024-11-03,danger_sign_alert,...,False,none,NaN,False,delivery,not_delivered_yet,SVD,none,NaN,respectful_care
1,C00002,Kiambu,Ruiru,Ruiru Health Centre,18,0,40,2024-01-28,2024-06-19,danger_sign_alert,...,True,Ruiru Sub-County Hospital,18.0,True,admitted,not_delivered_yet,NaN,none,4.0,lack_of_supplies
2,C00003,Machakos,Mavoko,Mavoko County Referral Hospital,11,2,43,2024-12-10,2025-01-05,education_message,...,False,none,NaN,False,referred_higher_level,not_delivered_yet,instrumental,none,NaN,respectful_care
3,C00004,Busia,Nambale,Nambale County Referral Hospital,24,2,34,2025-01-22,2025-02-11,danger_sign_alert,...,False,none,NaN,False,admitted,not_delivered_yet,SVD,none,NaN,respectful_care
4,C00005,Murang'a,Maragua,Maragua County Referral Hospital,30,5,42,2024-08-26,2024-09-20,education_message,...,False,none,NaN,False,referred_higher_level,not_delivered_yet,CS,none,NaN,lack_of_supplies


#### A. Core KPIs

##### 1. Reach and engagement

In [ ]:
# 1. Average number of clients per month

prompts_clients_df["interaction_date"] = pd.to_datetime(prompts_clients_df["interaction_date"], errors="coerce"
)

# Year–month
prompts_clients_df["interaction_year_month"] = prompts_clients_df["interaction_date"].dt.to_period("M")

# Active clients per month = distinct client_id per month
active_per_month = prompts_clients_df.groupby("interaction_year_month")["client_id"].nunique()

# Average number of active clients per month
average_active_clients_per_month = active_per_month.mean()

print(average_active_clients_per_month)

173.07692307692307


###### In the eight counties where the program operates, an average of 173 clients use PROMPTS each month to access maternal health services, indicating meaningful engagement but also highlighting room to scale outreach and enrollment further.